### ISSUE TITLE
allow sorting of annotation tables in session

### ISSUE THREAD

**Issue by vmt25** [pombase/canto#2310]
Sorting of annotation tables by genotype or term name during the curation session would be very useful for keeping track of annotations during curation.

A sorting feature is already available for the 'Admin' publication tables, so maybe the code can be easily re-used?

**Comment by kimrutherford** [comment]
From our call:
for phenotype annotation tables, sorting by genotype and term name would be useful.

![sort-annotation-tables-1](https://user-images.githubusercontent.com/90474/85399796-7f6b6080-b5ab-11ea-9e6e-e19840fbfea1.png)

@vmt25 which columns should be sortable for the interaction tables?  Interactor A, interactor B and the phenotype name?

![sort-annotation-tables-2](https://user-images.githubusercontent.com/90474/85400129-0b7d8800-b5ac-11ea-9ea9-689c0fbcb02d.png)

**Comment by vmt25** [comment]
> @vmt25 which columns should be sortable for the interaction tables? Interactor A, interactor B and the phenotype name?

Yes, those 3!

Speaking of which, it would be great if (non-single allele) genotypes could always show 'other'-type alleles before 'aberration' and 'accessory' alleles - see #2189

**Comment by ValWood** [comment]
Yes I would also like table sorting. I think we might have this in an older ticket? 
@mah11  do you remember?

**Comment by ValWood** [comment]
Yes I would also like table sorting. I think we might have this in an older ticket?
@mah11 do you remember?
I can't see an ticket. Maybe we only discussed reordering, but not flexible sorting...

**Comment by mah11** [comment]
I do vaguely recall chatting about it, but I don't remember whether it made it into a ticket.

**Comment by ValWood** [comment]
This would be useful for PHI canto too. We particularly want to be able to sort by "figure" is at the start of n the comment column. This lead to a discussion  about   splitting the "figure" into a separate column. We will open a ticket for this.

**Comment by kimrutherford** [comment]
I've started work on sorting annotation tables.  For now it just works for the Figure column and it's quite a hidden feature.  Clicking on "Figure" sorts the table.  An icon appears to show that it's sorted.  Click again to go back to normal sort order.  I've implemented only for the Figure column as a test.  I'll work on the other columns tomorrow, including the genotype column for FlyBase.

It currently sorts in alphabetical order which isn't great for Figure numbers (figure 11 would come after figure 1 and before figure 2 when sorted).

The change is available on the pombe main and test servers.

![sort-figure-1](https://user-images.githubusercontent.com/90474/88052828-f4cc5000-cbae-11ea-961a-a08954a6fa4d.png)

**Comment by ValWood** [comment]
Figure number above 7 is rare. In fact I have never seen it (most journals likely  have a lower limit). Above  this  they just put more figures in Supp, or split into Fig 1A,B,C,D,E,F

so that would probably work fine.

**Comment by kimrutherford** [comment]
> for phenotype annotation tables, sorting by genotype and term name would be useful.

Hi Vitor.

Sorting now works for the term name, term ID and extension columns in the phenotype annotation tables - click to the column heading to sort.

Sorting by genotype doesn't work yet.  You can click on the genotype column heading to change the order but the order you get doesn't make sense.  So the genotypes do get grouped together but not always in alphabetical order.  I fix that tomorrow and then I'll add sorting to the interaction tables.

**Comment by vmt25** [comment]
Already on the main site? Nice!

Thanks a million.

**Comment by kimrutherford** [comment]
> Sorting by genotype doesn't work yet.

That's working now so tomorrow I'll get on with fixing the sorting in the interaction tables.

**Comment by vmt25** [comment]
Awesome! Thanks!

**Comment by kimrutherford** [comment]
> That's working now so tomorrow I'll get on with fixing the sorting in the interaction tables.

That's done now.  If there are any other columns that you'd like to sort, please let me know.

**Comment by vmt25** [comment]
Yes, looks great, thanks.

The only other columns are the gene column and the curator column, right? If you want to be comprehensive and it is not too much work, then I guess it is fine to sorting by any columns. However, a) the gene column is a bit redundant with the genotype column and b) for FlyBase's purposes there is no present need to sort by curator because phenotype curation is done by a single curator.

**Comment by kimrutherford** [comment]
Thanks Vitor.  I've added sorting by curator and genes because it wasn't difficult.

> This would be useful for PHI canto too.

Alayne and Val could you let me know if you see any problems with sorting?

**Comment by ValWood** [comment]
works for me! (pombase canto)

**Comment by kimrutherford** [comment]
> works for me! (pombase canto)

Thanks for testing.  I think I'll close this issue.  We can open a new one for any bugs we find.

**Comment by CuzickA** [comment]
Hi @kimrutherford  and @jseager7 I can't see these changes yet in PHI-Canto. Please let me know when they are available for testing. Thanks.

**Comment by jseager7** [comment]
@CuzickA The sorting is already enabled, but maybe it's not obvious how to trigger it. You have to click the text in the column headings:

![image](https://user-images.githubusercontent.com/37659591/89163673-da5e8300-d56d-11ea-89d4-b2ee47ddd73c.png)

Then after clicking the text, an icon will appear to indicate which column is sorted:

![image](https://user-images.githubusercontent.com/37659591/89163758-fbbf6f00-d56d-11ea-8c4c-f4aa01ae7aca.png)

I've tested this on my test session on the production server and it works as expected.

We could make it more apparent how to sort the tables. Perhaps following something like Wikipedia, where there are dedicated sort buttons on each column header, which are always visible:

![image](https://user-images.githubusercontent.com/37659591/89163967-5eb10600-d56e-11ea-87d7-1a605f14a85e.png)

**Comment by kimrutherford** [comment]
From Skype: sort by species, host genotype and pathogen genotype.

**Comment by kimrutherford** [comment]
> From Skype: sort by species, host genotype and pathogen genotype.

Sort by species now works.  Sorting by genotypes will take a bit longer since it's trickier to implement.

**Comment by vmt25** [comment]
Working fine for FlyBase and no more work needed - from our standpoint, it can be closed
